### PR TITLE
Add landmarks to layout examples

### PR DESCRIPTION
**Pull request by lyubomir-popov**
## Done

Add landmarks to layout examples
Fixes https://github.com/canonical-web-and-design/vanilla-squad/issues/890

## QA

- Pull code
- Run `./run`
- Open the application examples [1](https://vanillaframework.io/docs/examples/layouts/application-structure), [2](https://vanillaframework.io/docs/examples/layouts/application) 
- Review use of landmarks (tags main, aside)

### CODE REVIEW

File: templates/docs/examples/layouts/application-structure.html [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
 
 {% block content %}
-<div class="l-application  app-demo">
+<div class="l-application app-demo" role="presentation">
   <div class="l-navigation-bar">
     <div class="u-clearfix"><code>l-navigation-bar</code> <button class="u-float-right js-menu-toggle is-dense u-no-margin">Menu</button></div>
   </div>
 
-  <div class="l-navigation">
+  <div class="l-navigation" role="banner">
```
Comment (bartaz) [inline templates/docs/examples/layouts/application-structure.html:10]: "By default, the HTML5 `<header>` element has an identical meaning to the banner landmark, unless it is a descendant of `<aside>`, `<article>`, `<main>`, `<nav>`, or `<section>`."

If we use HTML tags for other landmarks, maybe we should do the same for header/banner?

Comment (bartaz) [reply templates/docs/examples/layouts/application-structure.html:10]:
```suggestion
  <header class="l-navigation">
```

File: templates/docs/examples/layouts/application-structure.html [hunk 2]
```diff
@@ -17,7 +17,7 @@
     </div>
   </div>
```
Comment (bartaz) [inline templates/docs/examples/layouts/application-structure.html:18]:
```suggestion
  </header>
```

File: templates/docs/examples/layouts/application.html [hunk 3]
```diff
@@ -11,7 +11,7 @@
     </div>
   </div>
 
-  <div class="l-navigation is-collapsed">
+  <div class="l-navigation is-collapsed" role="banner">
```
Comment (bartaz) [inline templates/docs/examples/layouts/application.html:14]:
```suggestion
  <header class="l-navigation is-collapsed">
```

File: templates/docs/examples/layouts/application.html [hunk 4]
```diff
@@ -37,7 +37,7 @@
     </div>
   </div>
```
Comment (bartaz) [inline templates/docs/examples/layouts/application.html:38]:
```suggestion
  </header>
```

**Review by bartaz — DISMISSED** [review]
LGTM.

Small suggestion on using `<header>` in place for `role=banner`.

**Comment by bartaz** [comment]
One thing I noticed, we have another element in the layout `l-navigation-bar`. This is showing logo and menu button on mobile, so you can toggle side navigation.

Currently it doesn't have any role. Should it?
Semantically it's part of the navigation/header/banner, but it's a separate element, because it toggles the other one, and it itself is placed on the grid, so needs to be a direct child of `l-application`.

Screenshot 2020-09-11 at 13 39 05

**Review by bartaz — COMMENTED** [review]
I approved too soon, there is an open question about `l-navigation-bar`

**Comment by lyubomir-popov** [comment]
@bartaz this is a difficult question to resolve. Acdording to [the oinclusive components article](https://inclusive-components.design/menus-menu-buttons/#placement), the toggle needs to be inside the nav it toggles. But our structure won't allow that. I suggest tackling this as part of a separate task that looks at mobile toggles and updates them according to the article above.

**Review by bartaz — APPROVED** [review]
LGTM

**Pull request merged**